File: problems/2026-06-25/dsa-container-most-water/runner.py
import json as _json


def _run():
    _s = Solution()
    _cases = [
        (1, "standard example",   lambda: _s.maxArea([1, 8, 6, 2, 5, 4, 8, 3, 7]), 49),
        (2, "minimum two lines",  lambda: _s.maxArea([1, 1]),                         1),
        (3, "descending heights", lambda: _s.maxArea([4, 3, 2, 1]),                   4),
        (4, "equal heights",      lambda: _s.maxArea([5, 5, 5, 5]),                  15),
    ]
    _results = []
    for _id, _label, _call, _expected in _cases:
        try:
            _actual = _call()
        except Exception as _e:
            _actual = f"ERROR: {_e}"
        _results.append({
            "id": _id, "label": _label,
            "passed": _actual == _expected,
            "expected": _expected, "actual": _actual,
        })
    _passed = sum(1 for r in _results if r["passed"])
    print("__RESULT__:" + _json.dumps({"results": _results, "summary": {"passed": _passed, "total": len(_results)}}))


_run()

File: problems/2026-06-25/dsa-container-most-water/runner.ts
(function _run() {
  const _cases = [
    { id: 1, label: "standard example",   call: () => maxArea([1, 8, 6, 2, 5, 4, 8, 3, 7]), expected: 49 },
    { id: 2, label: "minimum two lines",  call: () => maxArea([1, 1]),                        expected: 1  },
    { id: 3, label: "descending heights", call: () => maxArea([4, 3, 2, 1]),                  expected: 4  },
    { id: 4, label: "equal heights",      call: () => maxArea([5, 5, 5, 5]),                  expected: 15 },
  ];
  const _results = _cases.map(c => {
    let _actual: unknown;
    try { _actual = c.call(); } catch (_e: any) { _actual = `ERROR: ${_e.message}`; }
    return { id: c.id, label: c.label, passed: JSON.stringify(_actual) === JSON.stringify(c.expected), expected: c.expected, actual: _actual };
  });
  const _passed = _results.filter(r => r.passed).length;
  process.stdout.write("__RESULT__:" + JSON.stringify({ results: _results, summary: { passed: _passed, total: _results.length } }) + "\n");
})();
